modules/crud/schema.go

```diff
@@ -0,0 +1,146 @@
+package crud
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/graphql-go/graphql/language/ast"
+	"github.com/graphql-go/graphql/language/kinds"
+	"github.com/graphql-go/graphql/language/parser"
+	"github.com/graphql-go/graphql/language/source"
+	"github.com/spaceuptech/space-cloud/config"
+	"github.com/spaceuptech/space-cloud/utils"
+)
+
+// ParseSchema Initializes Schema field in Module struct
+func (m *Module) ParseSchema(crud config.Crud) (SchemaType, error) {
+	schema := make(SchemaType, len(crud))
+	collection := SchemaCollection{}
+	for dbName, v := range crud {
+		for collectionName, v := range v.Collections {
+			source := source.NewSource(&source.Source{
+				Body: []byte(v.Schema),
+			})
+			// parse the source
+			doc, err := parser.Parse(parser.ParseParams{Source: source})
+			if err != nil {
+				return nil, err
+			}
+			value, _ := getSchemaDetails(doc, collectionName)
+			collection[collectionName] = value
+		}
+		schema[dbName] = collection
+	}
+	b, err := json.MarshalIndent(schema, "", "  ")
+	if err != nil {
+		fmt.Println("error:", err)
+	}
+	fmt.Print(string(b))
+	return schema, nil
+}
+
+func getSchemaDetails(doc *ast.Document, collectionName string) (SchemaField, error) {
+	fieldMap := SchemaField{}
+	for _, v := range doc.Definitions {
+		for _, ve := range v.(*ast.ObjectDefinition).Fields {
+
+			colName := v.(*ast.ObjectDefinition).Name.Value
+			// fmt.Println("Collection Name", collectionName, colName)
+
+			if colName != strings.Title(collectionName) {
+				continue
+			}
+			fieldTypeStuct := SchemaFieldType{
+				Directive: DirectiveProperties{
+					Value: DirectiveArgs{},
+				},
+			}
+
+			for _, w := range ve.Directives {
```

Comment:
Instead of the loop, check if the length of the `ve.Directives` array is more than zero. If yes, directly pick the 0th directive and work on that